src/cmd/top/summary_pane.c

```diff
@@ -0,0 +1,422 @@
+        x_length = 80;
+    if (res.used > res.total)
+        res.used = res.total;
+
+    snprintf (prefix, sizeof (prefix), "%5s [", name);
+    snprintf (suffix, sizeof (suffix), "%d/%d]", res.used, res.total);
+
+    int slots = x_length - strlen (prefix) - strlen (suffix) - 1;
+    mvwprintw (win,
+               y,
+               x,
+               "%s%*s%s",
+               prefix,
+               slots, "",
+               suffix);
+    /* Graph used */
+    wattron (win, COLOR_PAIR (TOP_COLOR_YELLOW));
+    for (int i = 0; i < ((double)res.used / res.total) * slots; i++)
+        mvwaddch (win, y, x + strlen (prefix) + i, '|');
+    wattroff (win, COLOR_PAIR (TOP_COLOR_YELLOW));
+
+    /* Graph down */
+    wattron (win, COLOR_PAIR (TOP_COLOR_RED));
+    for (int i = slots - 1; i >= slots - ((double)res.down / res.total) * slots; i--)
+        mvwaddch (win, y, x + strlen (prefix) + i, '|');
+    wattroff (win, COLOR_PAIR (TOP_COLOR_RED));
+}
+
+static void draw_resource (struct summary_pane *sum)
+{
+    draw_bargraph (sum->win,
+                   resource_dim.y_begin,
+                   resource_dim.x_begin,
+                   resource_dim.x_length,
+                   "nodes",
+                   sum->node);
+    draw_bargraph (sum->win,
+                   resource_dim.y_begin + 1,
+                   resource_dim.x_begin,
+                   resource_dim.x_length,
+                   "cores",
+                   sum->core);
+    draw_bargraph (sum->win,
+                   resource_dim.y_begin + 2,
+                   resource_dim.x_begin,
+                   resource_dim.x_length,
+                   "gpus",
+                   sum->gpu);
+}
+
+static void draw_heartbeat (struct summary_pane *sum)
+{
+    mvwprintw (sum->win,
+               heart_dim.y_begin,
+               heart_dim.x_begin,
+               "%s",
+               sum->heart_visible ? "♡" : " ");
+}
+
+/* Fetch expiration time (abs time relative to UNIX epoch) from resource.R.
+ * If unavailable (e.g. we are a guest in the system instance), return 0.
+ */
+static int get_expiration (flux_t *h, double *expiration)
+{
+    flux_future_t *f;
+    double val = 0;
+
+    if (!(f = flux_kvs_lookup (h, NULL, 0, "resource.R"))
+        || flux_kvs_lookup_get_unpack (f,
+                                       "{s:{s:f}}",
+                                       "execution",
+                                       "expiration", &val) < 0) {
+        if (errno == EPERM)
+            goto done;
+        log_err ("error fetching or decoding resource.R");
+        flux_future_destroy (f);
+        return -1;
+    }
+done:
+    flux_future_destroy (f);
+    *expiration = val;
+    return 0;
+}
+
+static int get_instance_level (flux_t *h, int *level)
+{
+    const char *s;
+    int lev = 0;
+
+    if (!(s = flux_attr_get (h, "instance-level"))) {
+        log_err ("error fetching instance-level broker attribute");
+        return -1;
+    }
+    lev = strtol (s, NULL, 10);
```

Comment:
should we check for parse error?  b/c there could be issues if `level` returns 0.